layer/filestore.go

```diff
@@ -346,8 +387,58 @@ func (fms *fileMetadataStore) List() ([]ChainID, []string, error) {
 	return ids, mounts, nil
 }
 
-func (fms *fileMetadataStore) Remove(layer ChainID) error {
-	return os.RemoveAll(fms.getLayerDirectory(layer))
+func (fms *fileMetadataStore) PrepareRemove(layer ChainID) (string, error) {
```

Comment:
Does this need to be exported?
I'm curious why you chose to do this preparation as a separate step from the actual remove. (separate as in caller manages this).

Reply:
Probably it does not. I was thinking that `deleteLayer` ends up more readable code like this and on some earlier version `Cleanup()` was sharing more code with it. I can move logic to `deleteLayer`....